Octokit/Http/ApiInfo.cs

```diff
@@ -51,5 +51,25 @@ public class ApiInfo
         /// Information about the API rate limit
         /// </summary>
         public RateLimit RateLimit { get; private set; }
+
+        /// <summary>
+        /// Allows you to clone ApiInfo 
+        /// </summary>
+        /// <returns>A clone of <seealso cref="ApiInfo"/></returns>
+        public ApiInfo Clone()
+        {
+            // Seem to have to do this to pass a whole bunch of tests (for example Octokit.Tests.Clients.EventsClientTests.DeserializesCommitCommentEventCorrectly)
+            // I believe this has something to do with the Mocking framework.
+            if (this == null || this.Links == null || this.OauthScopes == null || this.RateLimit == null || this.Etag == null)
+                return null;
```

Comment:
Extra points for explaining why I needed to add this null check.  For some reason bundles of tests fail without it (comment and see what happens).  I think it is something to do with how the mocking framework is handling the object - but can't understand why it touches it.